e2e/cypress/integration/enterprise/system_console/channel_modes_spec.js

```diff
@@ -8,6 +8,22 @@
 // ***************************************************************
 
 describe('Test channel public/private toggle', () => {
+    before(() => {
+        // Enable LDAP and LDAP group sync
+        cy.apiUpdateConfig({
+            LdapSettings: {Enable: true},
+            ServiceSettings: {ExperimentalLdapGroupSync: true},
+        });
+    });
+
+    after(() => {
+        // Disable LDAP and LDAP group sync
+        cy.apiUpdateConfig({
+            LdapSettings: {Enable: false},
+            ServiceSettings: {ExperimentalLdapGroupSync: false},
+        });
```

Comment:
Revert the config setting since it might affect other test, like when doing login via UI.

Reply:
Just one doubt: what if those settings where enabled before this test ran? Shouldn't we revert to what the settings were before starting with the test?

Reply:
Oppss sorry, it's actually not needed to revert since it's already handled in the corresponding test, like [here on login](https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/integration/login/login_spec.js#L19).

Re: your question, if required settings (all or some) are already set before this test, then no negative effect. It's still good; the only downside is we're setting it more than necessary.  In most cases, we're not reverting the changes to original state.  If a test requires a particular setting or state, then such test should do it on `before` or `beforeEach` hooks.    
See [related Cypress best practices](https://docs.cypress.io/guides/references/best-practices.html#Using-after-or-afterEach-hooks) for more info.